src/hera/workflows/_mixins.py

```diff
@@ -539,25 +539,25 @@ def _build_volume_mounts(self) -> Optional[List[VolumeMount]]:
         if self.volume_mounts is None and self.volumes is None:
             return None
 
-        if isinstance(self.volumes, list):
-            volumes = self.volumes
-        elif not isinstance(self.volumes, list) and self.volumes is not None:
-            volumes = [self.volumes]
-        else:
+        if self.volumes is None:
             volumes = []
+        else:
+            volumes = self.volumes if isinstance(self.volumes, list) else [self.volumes]
 
         result = (
             None
-            if self.volumes is None
+            if volumes is []
```

Comment:
Using `is` here is ringing alarm bells as it's only for `None` really (should use `==` or directly check on `volumes`). We could use a `not volumes` or swap the if/else around to use `if volumes`?
```suggestion
            if not volumes
```